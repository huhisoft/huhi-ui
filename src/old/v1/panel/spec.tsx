/* global jest, expect, describe, it, afterEach */
import * as React from 'react'
import { shallow } from 'enzyme'
import { create } from 'react-test-renderer'
import Panel from './index'

describe('panel tests', () => {
  const baseComponent = (props?: object) => (
    <Panel id='testPanel' {...props} />
  )
  describe('basic tests', () => {
    it('matches the snapshot', () => {
      const component = baseComponent()
      const tree = create(component).toJSON()
      expect(tree).toMatchSnapshot()
    })

    it('renders the component', () => {
      const wrapper = shallow(baseComponent({id: 'toughPanel'}))
      const assertion = wrapper.find('#toughPanel').length
      expect(assertion).toBe(1)
    })
  })

  describe('component behavior', () => {
    it('can have an id', () => {
      const wrapper = shallow(baseComponent({id: 'boxedThing'}))
      const assertion = wrapper.props().id
      expect(assertion).toBe('boxedThing')
    })
  })
})
