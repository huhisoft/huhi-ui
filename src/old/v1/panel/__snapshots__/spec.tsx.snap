// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`panel tests basic tests matches the snapshot 1`] = `
.c0 {
  padding: 40px;
  background-color: #FFF;
  font-size: inherit;
  box-sizing: border-box;
  border-radius: 20px;
  box-shadow: 0 6px 12px 0 rgba(39,46,64,0.2);
}

<div
  className="c0"
  id="testPanel"
/>
`;
